Refactor LSTM tests, [Remove randomness in weights]

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39920 Enable dynamic quantized GRU support
* #40102 Dynamic quantization support for LSTMCell, RNNCell and GRUCell [Remove randomness in weights]
* **#40101 Refactor LSTM tests, [Remove randomness in weights]**

Create three tests for LSTMs:
1. test_qlstm: Test to check numerics of quantized LSTM operator.
2. test_lstm_api: To check the LSTM module and compare
it with the quantized LSTM op
3. test_quantized_rnn: Check the dynamic quantization workflow, scriptability and serialization of quantized
LSTM

Differential Revision: [D22070826](https://our.internmc.facebook.com/intern/diff/D22070826/)

raghuramank10000 committed:
Refactor LSTM tests, [Remove randomness in weights]

Create three tests for LSTMs:
1. test_qlstm: Test to check numerics of quantized LSTM operator.
2. test_lstm_api: To check the LSTM module and compare
it with the quantized LSTM op
3. test_quantized_rnn: Check the dynamic quantization workflow, scriptability and serialization of quantized
LSTM

Differential Revision: [D22070826](https://our.internmc.facebook.com/intern/diff/D22070826/)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/test_quantized_module.py:
```diff
@@ -764,3 +763,47 @@ def test_linear_api(self, batch_size, in_features, out_features, use_bias, use_d
+            bias = True
+            bidirectional = False
+
+            x = torch.rand(seq_len, batch, input_size)
+            h = torch.rand(num_layers * (bidirectional + 1), batch, hidden_size)
+            c = torch.rand(num_layers * (bidirectional + 1), batch, hidden_size)
```
these are still random?

raghuramank100 replied:
I removed an empty affine quantized for the weight initialization. For the inputs, we do test with random values across our code. Shouldnt that be fine?

raghuramank10000 committed:
Update on "Refactor LSTM tests, [Remove randomness in weights]"


Create three tests for LSTMs:
1. test_qlstm: Test to check numerics of quantized LSTM operator.
2. test_lstm_api: To check the LSTM module and compare
it with the quantized LSTM op
3. test_quantized_rnn: Check the dynamic quantization workflow, scriptability and serialization of quantized
LSTM

Differential Revision: [D22070826](https://our.internmc.facebook.com/intern/diff/D22070826/)

[ghstack-poisoned]
